src/content/marina/marinaBroker.ts

```diff
@@ -326,6 +328,102 @@ export default class MarinaBroker extends Broker {
           return successMsg();
         }
 
+        case Marina.prototype.broadcastTransaction.name: {
+          this.checkHostnameAuthorization(state);
+          const network = selectNetwork(state);
+          const [signedTxHex] = params as [string];
+
+          // broadcast tx
+          const txid = await broadcastTx(selectEsploraURL(state), signedTxHex);
+          if (!txid) throw new Error('something went wrong with the tx broadcasting');
+
+          // will need tx to lock selected utxos and credit unconfirmed utxos
+          const tx = Transaction.fromHex(signedTxHex);
+
+          // lock selected utxos used in this transaction:
+          // - get all coins from marina
+          // - iterate over all transaction inputs and check if is a coin of ours
+          // - if any, lock selected utxos
+
+          // get all coins from marina
+          const coins = selectUtxos(MainAccountID)(state);
+
+          // array to store all utxos selected in this transaction
+          const selectedUtxos: UnblindedOutput[] = [];
+
+          // find all inputs that are a coin of ours
+          tx.ins.forEach((_in) => {
+            const { hash, index } = _in; // txid and vout
+            const txid = hash.reverse().toString('hex');
+            for (const coin of coins) {
+              if (coin.txid === txid && coin.vout === index) {
+                selectedUtxos.push(coin);
+                break;
+              }
+            }
+          });
+
+          // lock utxos used in this transaction
+          if (selectedUtxos && selectedUtxos.length > 0) {
+            for (const utxo of selectedUtxos) {
+              await this.store.dispatchAsync(lockUtxo(utxo));
+            }
+          }
+
+          // credit change utxos:
+          // - get all addresses used by marina
+          // - iterate over all transaction outputs and check if belongs to marina
+          // - if any, add unconfirmed utxos to utxo set
+
+          // get all marina addresses (reversed for performance reasons)
+          const xpub = await selectMainAccount(state).getWatchIdentity(network);
```

Comment:
It's ok now but ideally it should work with multiple accounts "paradigm". Maybe u could select all accounts' here ?

Reply:
Fixed by 7244ddb